docs/appendix/zowe-glossary.md

```diff
@@ -0,0 +1,300 @@
+
+* **Extension**
+
+  An extension is generally used to describe additional, non-default Zowe plugins or components.
+
+* **Core components**
+
+  The definition of a core component is governed by the TSC but a core component usually means a packaged, foundational piece as part of base Zowe. From the perspective of a conformant support provider, providing support for Zowe refers to providing support for each core component of Zowe though a provider may put their own limitations on what they support. A core component is usually actively maintained by one or more squads. A component has a component manifest file, that helps identify it with the rest of Zowe. See: https://docs.zowe.org/stable/appendix/server-component-manifest/
+
+## Core
+
+* **Zowe server components**
+
+  Also known as "Zowe z/OS components" or "Zowe server-side components". Includes all the Zowe components that are installed on z/OS. 
+
+* **Zowe client components**
+ 
+  Also known as "Zowe client-side components". Includes all the Zowe components that are installed on the users' PC. 
+
+* **API Mediation Layer (API ML)**
+
+  A core Zowe component that provides a reverse proxy and enables REST APIs by providing a single point of access for mainframe service REST APIs like MVS Data Sets, JES, as well as working with z/OSMF. It has dynamic discovery capability for these services and Gateway is also responsible for generating the authentication token used to provide single sign-on (SSO) functionality.
+
+* **Zowe Application Framework**
+
+  A core Zowe component that modernizes and simplifies working on the mainframe via a web visual interface. Functionality is provided through apps and a desktop user experience called the Zowe Desktop. Base functionality includes apps to work with JES, MVS Data Sets, Unix System Services, as well as a 3270 Terminal, Virtual Terminal, and an Editor.
+
+* **Zowe CLI**
+
+  A core Zowe component that provides a command-line interface that lets you interact with the mainframe remotely and use common tools such as Integrated Development Environments (IDEs), shell commands, bash scripts, and build tools for mainframe development. The core set of commands includes working with data sets, USS, JES, as well as issuing TSO and console commands. The Zowe CLI is incredibly popular in modern mainframe education.
+
+* **Zowe Explorer** 
+
+  A core Zowe component that is a Visual Studio Code extension that modernizes the way developers and system administrators interact with z/OS mainframes. The Zowe Explorer lets you interact with data sets, USS files, and jobs that are stored on z/OS. The Zowe Explorer is incredibly popular in modern mainframe education.
+
+* **Zowe Client SDKs** 
+
+  Client SDK's allow extenders to build applications on top of existing programmatic APIs such as z/OSMF. Currently supported client SDK's include Node.js (core), Kotlin/z/OSMF, Python, Swift, and Java. 
```

Comment:
Only the Node.js SDK is part of Core. The rest are extensions.

Reply:
@balhar-jakub I wrote that in the second sentence, but I could think about how to make it clearer

Reply:
I restructured the SDK related glossary part in my updates.